beamer/check/commands.py

```diff
@@ -478,3 +478,246 @@ def verify_l1_invalidations(
+    argument_filters = dict(requestId=challenge.request_id)
+    log.debug(
+        "Waiting for agent to claim",
+        chain_id=challenge.request_chain,
+        request_id=challenge.request_id,
+    )
+    while True:
+        events = request_manager.events.ClaimMade.get_logs(  # type: ignore
+            fromBlock=from_block, argument_filters=argument_filters
+        )
+        if events:
+            assert len(events) == 1
+            break
+        time.sleep(1)
+
+    event = events[0]
+    challenge.claim_id = ClaimId(event.args["claimId"])
+    challenge.txhash_claimRequest = event.transactionHash
+
+
+def _create_transfer_request(
+    ctx: Context, request_chain: ChainId, fill_chain: ChainId, symbol: str, target_token: Contract
+) -> Challenge:
+    request_manager = _obtain_request_manager(ctx, request_chain)
+    validity_period = request_manager.functions.MIN_VALIDITY_PERIOD().call()
+
+    source_token = _find_token(ctx, request_chain, symbol)
+    if source_token is None:
+        log.error("Could not find token", token=symbol, chain_id=request_chain)
+        sys.exit(1)
+
+    _transact(source_token.functions.approve(request_manager.address, 1))
+
+    func = request_manager.functions.createRequest(
+        fill_chain,
+        source_token.address,
+        target_token.address,
+        ctx.account.address,
+        TokenAmount(1),
+        validity_period,
+    )
+    receipt = _transact(func)
+    event = request_manager.events.RequestCreated().process_log(receipt["logs"][0])
+    request_id = RequestId(event.args["requestId"])
+    txhash = receipt.transactionHash.hex()  # type: ignore
+    log.info(
+        "Created transfer request",
+        request_chain=request_chain,
+        fill_chain=fill_chain,
+        txhash=txhash,
+    )
+
+    return Challenge(
+        request_chain=request_chain,
+        fill_chain=fill_chain,
+        request_id=request_id,
+        claim_id=None,
+        txhash_createRequest=HexBytes(txhash),
+        txhash_claimRequest=None,
+        txhash_challengeClaim=None,
+        finalization_timestamp=None,
+    )
+
+
+def _challenge_claim(ctx: Context, challenge: Challenge, stake: float) -> None:
+    request_manager = _obtain_request_manager(ctx, challenge.request_chain)
+    func = request_manager.functions.challengeClaim(challenge.claim_id)
+    receipt = _transact(func, value=to_wei(stake, "ether"))
+    txhash = receipt.transactionHash.hex()  # type: ignore
+    log.info(
+        "Challenged claim",
+        chain_id=challenge.request_chain,
+        claim_id=challenge.claim_id,
+        txhash=txhash,
+    )
+
+    block = request_manager.w3.eth.get_block(receipt["blockNumber"])
+    finality_period = request_manager.functions.chains(challenge.fill_chain).call()[0]
+    timestamp = datetime.utcfromtimestamp(block["timestamp"])
+    challenge.txhash_challengeClaim = HexBytes(txhash)
+    challenge.finalization_timestamp = timestamp + timedelta(seconds=finality_period)
```

Comment:
i think there is a problem here related with OP chains

Reply:
> i think there is a problem here related with OP chains

Could you please be more specific?

Reply:
After a lengthy discussion, we decided to go with the current finalization timestamp computation. Ideally, we would start the finalization period from the moment the proof is executed on L1 (this is done by the agent internally), however, it is not straightforward to get the proof tx timestamp and possible solutions (e.g. watching for events on L1, querying the related OP contracts) would add much complexity.